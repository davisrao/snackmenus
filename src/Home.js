import React from "react";
import { Card, CardBody, CardTitle } from "reactstrap";

/** HomePage component
 * props:
 * - total drinks, total snacks (counts of total in DB)
 * 
 * state:
 * - none
 * 
 * context:
 * - none
 * 
 * HomePage => {}
 */

function Home({ totalDrinks, totalSnacks }) {
  console.log("made it to home page")
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h3 className="font-weight-bold">
              Welcome to your favorite dive cafe!
            </h3>
            <h5>{`Home of ${totalDrinks} drink and ${totalSnacks} snack options.`}</h5>
          </CardTitle>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
