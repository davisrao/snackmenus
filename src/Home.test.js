import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home"



//run tests with npm test in terminal

describe("ErrorMessage Test", function () {
    it("matches snapshot", function () {
        const result = render(
            <Home totalDrinks={1} totalSnacks={2}/>
        );
        expect(result).toMatchSnapshot();
    });
    it("renders with expected text", function () {
        const result = render(
            <Home totalDrinks={1} totalSnacks={2}/>
        );

        expect(result.queryByText("Welcome to your favorite dive cafe!")).toBeInTheDocument();

    });
});