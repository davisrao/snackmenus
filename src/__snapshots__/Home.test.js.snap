// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Test matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="col-md-8"
      >
        <div
          class="card"
        >
          <div
            class="text-center card-body"
          >
            <div
              class="card-title"
            >
              <h3
                class="font-weight-bold"
              >
                Welcome to your favorite dive cafe!
              </h3>
              <h5>
                Home of 1 drink and 2 snack options.
              </h5>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="col-md-8"
    >
      <div
        class="card"
      >
        <div
          class="text-center card-body"
        >
          <div
            class="card-title"
          >
            <h3
              class="font-weight-bold"
            >
              Welcome to your favorite dive cafe!
            </h3>
            <h5>
              Home of 1 drink and 2 snack options.
            </h5>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
